goinp/goinp.go

```diff
@@ -120,37 +120,47 @@ func AskOptions(title string, defaultValue string, optional bool, options ...str
 	}
 
 	for i, option := range options {
-		log.Printf("[%d] : %s", i+1, option)
+		fmt.Printf("[%d] : %s\n", i+1, option)
 	}
 
 	for {
 		fmt.Print("Type in the option's number, then hit Enter: ")
 
-		answer, err := bufio.NewReader(os.Stdin).ReadString('\n')
+		answer, err := reader.ReadString('\n')
 		if err != nil {
-			log.Errorf("failed to read input value")
+			fmt.Printf(colorstring.Red("failed to read input value") + "\n")
 			continue
 		}
 
 		optionNo, err := strconv.Atoi(strings.TrimSpace(answer))
 		if err != nil {
-			log.Errorf("failed to parse option number, pick a number from 1-%d", len(options))
+			fmt.Print(colorstring.Red(fmt.Sprintf("failed to parse option number, pick a number from 1-%d", len(options))) + "\n")
 			continue
 		}
 
 		if optionNo-1 < 0 || optionNo-1 >= len(options) {
-			log.Errorf("invalid option number, pick a number 1-%d", len(options))
+			fmt.Print(colorstring.Red(fmt.Sprintf("invalid option number, pick a number 1-%d", len(options))) + "\n")
 			continue
 		}
 
-		if options[optionNo-1] == customValueOptionText {
-			return AskOptions(title, "", true)
+		if optionNo == len(options) && optional {
+			return askOptionsWithReader(title, "", true, reader)
 		}
 
 		return options[optionNo-1], nil
 	}
 }
 
+// AskOptions ...
+func AskOptions(title string, defaultValue string, optional bool, options ...string) (string, error) {
+	reader := bufio.NewReader(os.Stdin)
+
+	if len(options) == 0 {
```

Comment:
Is not this the same as calling `return askOptionsWithReader(title, defaultValue, optional, reader, options...)` with the empty options?

Reply:
Sure, it is.

Reply:
Also, reader is not OK like this, bacause *os.File is converted into io.Reader which is not writable at all and there is no option to convert back.